2007-08-20  Corentin Le Gall <clegall@proformatique.com>

	pf_voicemail_date_language.patch
	$Revision$
	$Date$

	The date language is appropriately set in the main body of
	voicemail-issued e-mails.

Index: asterisk-1.4.24/apps/app_voicemail.c
===================================================================
--- asterisk-1.4.24.orig/apps/app_voicemail.c	2009-03-06 19:23:09.000000000 +0100
+++ asterisk-1.4.24/apps/app_voicemail.c	2009-05-29 19:34:10.000000000 +0200

@@ -65,6 +65,9 @@
 #include <sys/types.h>
 #include <sys/mman.h>
 #include <time.h>
+#ifdef HAVE_LOCALE_H
+#include <locale.h>
+#endif
 #include <dirent.h>
 #ifdef IMAP_STORAGE
 #include <ctype.h>
@@ -581,6 +584,11 @@
 static unsigned char adsifdn[4] = "\x00\x00\x00\x0F";
 static unsigned char adsisec[4] = "\x9B\xDB\xF7\xAC";
 static int adsiver = 1;
+
+#ifdef HAVE_LOCALE_H
+static char emaildatelocale[32];
+AST_MUTEX_DEFINE_STATIC(setlocale_emaildate_lock);
+#endif
 static char emaildateformat[32] = "%A, %B %d, %Y at %r";
 
 
@@ -3164,7 +3172,21 @@
 	fprintf(p, "Date: %s" ENDL, date);
 
 	/* Set date format for voicemail mail */
+#ifdef HAVE_LOCALE_H
+	if (*emaildatelocale) {
+		char *loc;
+		ast_mutex_lock(&setlocale_emaildate_lock);
+		loc = ast_strdup(setlocale(LC_TIME, NULL));
+		setlocale(LC_TIME, emaildatelocale);
+		strftime(date, sizeof(date), emaildateformat, &tm);
+		setlocale(LC_TIME, loc);
+		ast_free(loc);
+		ast_mutex_unlock(&setlocale_emaildate_lock);
+	} else
+		strftime(date, sizeof(date), emaildateformat, &tm);
+#else
 	strftime(date, sizeof(date), emaildateformat, &tm);
+#endif
 
 	if (*fromstring) {
 		struct ast_channel *ast;
@@ -8298,6 +8320,14 @@
 			ast_copy_string(emaildateformat, emaildateformatstr, sizeof(emaildateformat));
 		}
 
+#ifdef HAVE_LOCALE_H
+		/* Load date locale config for voicemail mail */
+		memset(emaildatelocale,0,sizeof(emaildatelocale));
+		if ((s = ast_variable_retrieve(cfg, "general", "emaildatelocale"))) {
+			ast_copy_string(emaildatelocale,s,sizeof(emaildatelocale));
+		}
+#endif
+
 		/* External password changing command */
 		if ((extpc = ast_variable_retrieve(cfg, "general", "externpass"))) {
 			ast_copy_string(ext_pass_cmd,extpc,sizeof(ext_pass_cmd));
